inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/util/QueryUtil.java

```diff
@@ -67,58 +68,146 @@
     private static final String WIKIMEDIA_NEWS_ARTICLE = "e:Q17633526";
 
     private static final String WIKIMEDIA_NAVIGATIONAL_TEMPLATE = "e:Q11753321";
+
+    private static String getExactMatchingQueryPart(String aString)
+    {
+        return String.join("\n",
+            "    SELECT DISTINCT ?e2 ?description WHERE",
+            "    {",
+            "     VALUES ?labelpredicate {rdfs:label skos:altLabel}",
+            "      {",
+            "        ?e2 ?labelpredicate ?" + aString + " @en .",
```

Comment:
This opens the way to query injection problems. If the query string includes a double quote, the SPARQL query breaks. I see that some escaping is done in other locations - but it would be better to do the escaping where it is needed, namely when constructing the query.

Reply:
Is it actually necessary to inject the query string in this way? Isn't it possible to create a string literal with a language and bind that to a proper query variable here?

Reply:
I inject it here so I don't have to write the query part twice. But you are right about creating the literal with a language.

Reply:
> but it would be better to do the escaping where it is needed, namely when constructing the query.

I have double quotes in the query (`FILTER ( lang(?label) = "en")`). If I escape when constructing the query (`TupleQuery tupleQuery = conn
            .prepareTupleQuery(QueryLanguage.SPARQL, RenderUtils.escape(query)); `), the backslashes are not accepted by the SPARQL processor.

Reply:
I am calling this query twice, once with the surface form and once with the typed string. I made that part variable so I don't have to write duplicate code.

Reply:
Ok, but that doesn't require that you change the query. I'd just use a hard-coded query variable like "?value" and in one case bind the surface form to it and in the other the typed string.